apps/server/src/modules/authentication/services/account.service.spec.ts

```diff
@@ -0,0 +1,233 @@
+import { MikroORM, NotFoundError } from '@mikro-orm/core';
+import { Test, TestingModule } from '@nestjs/testing';
+import { EntityNotFoundError } from '@shared/common';
+import { Account, EntityId, Role, School, User } from '@shared/domain';
+import { AccountRepo, SystemRepo, UserRepo } from '@shared/repo';
+import { accountFactory, schoolFactory, setupEntities, userFactory } from '@shared/testing';
+import { AccountEntityToDtoMapper } from '../mapper/account-entity-to-dto.mapper';
+import { AccountService } from './account.service';
+import { AccountDto } from './dto/account.dto';
+
+describe('AccountService', () => {
+	let module: TestingModule;
+	let accountService: AccountService;
+	let orm: MikroORM;
+	let mockAccounts: Account[];
+	let mockUsers: User[];
+	let accountRepo: AccountRepo;
+
+	const defaultPasswordHash = '$2a$10$/DsztV5o6P5piW2eWJsxw.4nHovmJGBA.QNwiTmuZ/uvUc40b.Uhu';
+
+	let mockSchool: School;
+
+	let mockTeacherUser: User;
+	let mockStudentUser: User;
+	let mockUserWithoutAccount: User;
+
+	let mockTeacherAccount: Account;
+	let mockStudentAccount: Account;
+	const userRepoFindMock = jest.fn().mockImplementation((userId) => mockUsers.find((user) => user.id === userId));
```

Comment:
Jus to avoid mixing concepts, we may move that to the modules definition and expect on the userRepo instead. 

`	
	userRepo = module.get(UserRepo); 
	...
	expect(userRepo.findById).toHaveBeenCalledWith(mockTeacherUser.id);
`